assets/src/settings-page/index.js

```diff
@@ -135,7 +136,7 @@ function Root( { appRoot } ) {
 		};
 	}, [ fetchingOptions ] );
 
-	if ( false !== fetchingOptions ) {
+	if ( false !== fetchingOptions || null === templateModeWasOverridden ) {
```

Comment:
Instead of `templateModeWasOverridden` maybe it would make more sense to use `fetchingThemes` here?

Reply:
I guess not since that's not sufficient to prevent the re-render.

Reply:
Why not just check if `fetchingOptions` is truthy?

```suggestion
	if ( fetchingOptions || null === templateModeWasOverridden ) {
```

Reply:
We want to show the loading spinner not only when the options are being fetched (`true`), but also initially, when the state is not "stable" yet (`null`).

Although now with the second condition this case is covered via `templateModeWasOverriden`, I'd still keep the explicit `false !== fetchingOptions` in here.